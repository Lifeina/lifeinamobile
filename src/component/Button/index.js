import React from 'react'
import PropTypes from 'prop-types'
import { Button } from '@ui-kitten/components'
import { StyleSheet } from 'react-native'
import { Text } from '@ui-kitten/components'

const customButton = (props) => {
  const { children, onPress } = props

  return (
    <Button style={styles.button} onPress={onPress}>
      <Text styles={styles.textBtn}>{children}</Text>
    </Button>
  )
}

const styles = StyleSheet.create({
  button: {
    marginVertical: 16,
    backgroundColor: 'white',
    borderColor: 'white'
  },
  textBtn: {
    textAlign: 'center'
  }
})

customButton.propTypes = {
  children: PropTypes.node.isRequired,
  onPress: PropTypes.func,
  accessoryLeft: PropTypes.func
}

export { customButton as Button }
