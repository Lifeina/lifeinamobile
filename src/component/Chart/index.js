import React from 'react'
import { View, StyleSheet } from 'react-native'
import {
  VictoryLine,
  VictoryChart,
  VictoryZoomContainer,
  VictoryLegend
} from 'victory-native'
import { theme } from './theme'

const Chart = ({ chartValue }) => {
  return (
    <View style={styles.container}>
      <VictoryChart
        domain={{ y: [0, 110] }}
        padding={70}
        height={350}
        theme={theme}
        animate={{ duration: 200 }}
        domainPadding={20}
        containerComponent={
          <VictoryZoomContainer
            zoomDomain={{ y: [0, 110] }}
            responsive
            zoomDimension="x"
          />
        }>
        <VictoryLegend
          x={155}
          y={0}
          title="Battery"
          centerTitle
          orientation="horizontal"
          gutter={20}
          style={{ border: { stroke: 'black' }, title: { fontSize: 20 } }}
          data={[
            {
              name: `Charge ${chartValue[0].batteryValue}%`,
              symbol: { fill: 'white' }
            }
          ]}
        />
        <VictoryLine
          style={{
            data: { stroke: 'white', strokeWidth: 2 },
            parent: { border: '1px solid #ccc' }
          }}
          data={chartValue}
          x="timeElapsed"
          y="batteryValue"
        />
      </VictoryChart>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 'auto'
  }
})

Chart.defaultProps = {
  chartValue: [
    { timeElapsed: 0, batteryValue: 100 },
    { timeElapsed: -15, batteryValue: 90 },
    { timeElapsed: -30, batteryValue: 80 },
    { timeElapsed: -45, batteryValue: 70 },
    { timeElapsed: -60, batteryValue: 60 },
    { timeElapsed: -75, batteryValue: 50 },
    { timeElapsed: -90, batteryValue: 40 },
    { timeElapsed: -105, batteryValue: 30 },
    { timeElapsed: -120, batteryValue: 20 },
    { timeElapsed: -135, batteryValue: 10 },
    { timeElapsed: -150, batteryValue: 1 }
  ]
}

export { Chart }
