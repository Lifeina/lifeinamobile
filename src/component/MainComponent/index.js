import React from 'react'
import * as eva from '@eva-design/eva'
import { store } from '../../store'
import { StoreProvider } from 'easy-peasy'
import { ApplicationProvider, IconRegistry } from '@ui-kitten/components'
import { EvaIconsPack } from '@ui-kitten/eva-icons'
import { mapping, light } from '@eva-design/eva'
import { default as appTheme } from '../../assets/custom-theme.json'
import PropTypes from 'prop-types'

const MainComponent = ({ children }) => (
  <StoreProvider store={store}>
    <IconRegistry icons={EvaIconsPack} />
    <ApplicationProvider
      mapping={mapping}
      {...eva}
      theme={{ ...light, ...appTheme }}>
      {children}
    </ApplicationProvider>
  </StoreProvider>
)

MainComponent.propTypes = {
  children: PropTypes.node.isRequired
}

export { MainComponent }
