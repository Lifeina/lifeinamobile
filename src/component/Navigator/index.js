import React, { useState } from 'react'
import { StyleSheet } from 'react-native'
import {
  BottomNavigation,
  BottomNavigationTab,
  Icon
} from '@ui-kitten/components'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import { HomeView, SettingsView, ChartsView } from '../../views'

const BottomTab = createBottomTabNavigator()

const BottomTabBar = ({ navigation, state }) => {
  const [selectedIndex, setSelectedIndex] = useState(0)
  const HomeIcon = (style) => <Icon {...style} name="home-outline" />
  const ChartsIcon = (style) => <Icon {...style} name="bar-chart-outline" />
  const SettingsIcon = (style) => <Icon {...style} name="settings-2-outline" />

  const onSelect = (index) => {
    setSelectedIndex(index)
    navigation.navigate(state.routeNames[index])
  }

  return (
    <BottomNavigation
      style={styles.bottomNavigation}
      indicatorStyle={styles.indicator}
      selectedIndex={selectedIndex}
      onSelect={onSelect}>
      <BottomNavigationTab icon={HomeIcon} title="Home" />
      <BottomNavigationTab icon={ChartsIcon} title="Charts" />
      <BottomNavigationTab icon={SettingsIcon} title="Settings" />
    </BottomNavigation>
  )
}

const TabNavigator = () => {
  return (
    <BottomTab.Navigator tabBar={(props) => <BottomTabBar {...props} />}>
      <BottomTab.Screen
        name="Home"
        style={styles.container}
        component={HomeView}
      />
      <BottomTab.Screen
        name="Charts"
        style={styles.container}
        component={ChartsView}
      />
      <BottomTab.Screen
        name="Settings"
        style={styles.container}
        component={SettingsView}
      />
    </BottomTab.Navigator>
  )
}

const styles = StyleSheet.create({
  bottomNavigation: { backgroundColor: 'white' },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 35
  },
  spinner: {
    height: 100,
    display: 'flex',
    justifyContent: 'center'
  }
})

export { TabNavigator }
