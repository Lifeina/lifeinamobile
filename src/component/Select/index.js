import React, { useState } from 'react'
import { IndexPath, Select, SelectItem } from '@ui-kitten/components'
import { StyleSheet } from 'react-native'
const CustomSelect = ({ data }) => {
  const [selectedIndex, setSelectedIndex] = useState(new IndexPath(0))

  const displayValue = data[selectedIndex.row]

  const renderOption = (title, index) => (
    <SelectItem key={index} title={title} />
  )

  return (
    <Select
      value={displayValue}
      selectedIndex={selectedIndex}
      onSelect={(index) => {
        // console.log(index)
        setSelectedIndex(index)
      }}
      style={styles.select}>
      {data.map(renderOption)}
    </Select>
  )
}

const styles = StyleSheet.create({
  select: { width: '100%', textAlign: 'center' }
})

export { CustomSelect as Select }
