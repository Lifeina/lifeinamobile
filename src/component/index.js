export * from './Button'
export * from './Select'
export * from './Chart'
export * from './MainComponent'
export * from './Navigator'
