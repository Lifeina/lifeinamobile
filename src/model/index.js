import temperatureModel from './temperature-model'
import batteryModel from './battery-model'
import settingsModel from './settings-model'
import deviceModel from './device-model'
import languagesModel from './languages-model'

const storeModel = {
  temperature: temperatureModel,
  battery: batteryModel,
  settings: settingsModel,
  device: deviceModel,
  translation: languagesModel
}

export default storeModel
