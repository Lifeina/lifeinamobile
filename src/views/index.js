export * from './Home'
export * from './Charts'
export * from './Settings'
