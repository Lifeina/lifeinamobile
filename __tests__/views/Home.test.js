import 'react-native'
import React from 'react'
import { HomeView } from '../../src/views'
import { MainComponent } from '../../src/component'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders Home view correctly', () => {
  expect(
    renderer.create(
      <MainComponent>
        <HomeView />
      </MainComponent>
    )
  ).toMatchSnapshot()
})
