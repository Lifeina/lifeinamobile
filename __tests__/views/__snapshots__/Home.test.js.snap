// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home view correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(54, 188, 238)",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/logo-lifeina.png",
        }
      }
      style={
        Object {
          "height": 108,
          "marginBottom": 30,
          "width": 102,
        }
      }
    />
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          undefined,
        ]
      }
    />
    <Text
      appearance="default"
      category="h3"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 30,
            "fontWeight": "800",
          },
          Object {
            "color": "white",
            "marginBottom": 30,
          },
        ]
      }
    >
      Welcome to Lifeina
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 16,
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 10,
            },
          ]
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
          styles={
            Object {
              "textAlign": "center",
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            FIND MY LIFEINABOX
          </Text>
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
